presto-main/src/main/java/com/facebook/presto/cost/FilterStatsCalculator.java

```diff
@@ -175,6 +185,58 @@ protected PlanNodeStatsEstimate visitIsNullPredicate(IsNullPredicate node, Void
             return visitExpression(node, context);
         }
 
+        @Override
+        protected PlanNodeStatsEstimate visitBetweenPredicate(BetweenPredicate node, Void context)
+        {
+            if (!(node.getValue() instanceof SymbolReference) || !(node.getMin() instanceof Literal) || !(node.getMax() instanceof Literal)) {
+                return visitExpression(node, context);
+            }
+
+            SymbolStatsEstimate valueStats = input.getSymbolStatistics(Symbol.from(node.getValue()));
+            Expression leftComparison;
+            Expression rightComparison;
+
+            // We want to do heuristic cut (infinite range to finite range) ASAP and than do filtering on finite range.
+            if (isInfinite(valueStats.getLowValue())) {
+                leftComparison = new ComparisonExpression(GREATER_THAN_OR_EQUAL, node.getValue(), node.getMin());
+                rightComparison = new ComparisonExpression(LESS_THAN_OR_EQUAL, node.getValue(), node.getMax());
+            }
+            else {
+                rightComparison = new ComparisonExpression(GREATER_THAN_OR_EQUAL, node.getValue(), node.getMin());
+                leftComparison = new ComparisonExpression(LESS_THAN_OR_EQUAL, node.getValue(), node.getMax());
+            }
+
+            // we relay on and processing left to right
+            return process(and(leftComparison, rightComparison));
+        }
+
+        @Override
+        protected PlanNodeStatsEstimate visitInPredicate(InPredicate node, Void context)
+        {
+            if (!(node.getValueList() instanceof InListExpression) || !(node.getValue() instanceof SymbolReference)) {
+                return visitExpression(node, context);
+            }
+
+            InListExpression inList = (InListExpression) node.getValueList();
+            PlanNodeStatsEstimate statsSum = inList.getValues().stream()
```

Comment:
why doesn't this also subtract the AND of the inList predicates like with OR?  (relevant e.g. if you specify duplicate values)